### PR TITLE
dbus api: Handle deferred at start up

### PR DESCRIPTION
**Pull request by tasleson**
When api::connect is executed the code iterates over the engine processing
all the objects.  It makes calls to create_dbus_pool, create_dbus_filesystem,
and create dbus_block_dev.  The problem is all of these functions add the
changes to the "action" queue.  Thus when then the first client does any
method call it will appear the system is empty until after we process
the deferred event queue after handling the initial request.

This change moves the deferred processing into a separate function and
calls it in api::connect before returning, so that the tree has all the
queued objects applied to it.

Signed-off-by: Tony Asleson <tasleson@redhat.com>

Resolves: #627

### CODE REVIEW

File: src/dbus_api/api.rs [hunk 1]
```diff
@@ -282,9 +282,31 @@ pub fn connect(engine: Rc<RefCell<Engine>>)
 
     c.register_name(STRATIS_BASE_SERVICE, NameFlag::ReplaceExisting as u32)?;
 
+    process_deferred_actions(&c, &mut tree, &dbus_context)?;
+
     Ok((c, tree, dbus_context))
 }
 
+fn process_deferred_actions(c: &Connection,
```
Comment (mulkieran) [inline src/dbus_api/api.rs]: Please put header comment on this method.

File: src/dbus_api/api.rs [hunk 2]
```diff
@@ -299,19 +321,8 @@ pub fn handle(c: &Connection,
...
-                    c.register_object_path(path.get_name())?;
-                    tree.insert(path);
-                }
-                DeferredAction::Remove(path) => {
-                    c.unregister_object_path(&path);
-                    tree.remove(&path);
-                }
-            }
-        }
+        // Take all adds/removes from tree and handle them
```
Comment (mulkieran) [inline src/dbus_api/api.rs]: Please drop this comment.

File: src/dbus_api/api.rs (hunk 1 above)
Comment (mulkieran) [inline src/dbus_api/api.rs]: Are you able to pass only the actions, not the whole context.?

Comment (tasleson) [reply src/dbus_api/api.rs]: I tried, but after spending way too much time I gave up.

File: src/dbus_api/api.rs (hunk 2 above)
Comment (tasleson) [reply src/dbus_api/api.rs]: Done

File: src/dbus_api/api.rs (hunk 1 above)
Comment (tasleson) [reply src/dbus_api/api.rs]: Done

**Comment by agrover** [comment]
@tasleson I pushed a commit to pass actions instead of entire DbusContext, please take a look.

**Comment by tasleson** [comment]
@agrover LGTM, Thanks!

**Comment by agrover** [comment]
ok cool. If you'd be so kind as to squash it into the previous commit, we may be ready for final review.

**Comment by tasleson** [comment]
@agrover Commit squashed and I re-based, so hopefully this will pass CI with a new updated clippy.

**Comment by agrover** [comment]
👍 , merging even though clippy is broken. Thanks!

**Pull request merged**